src/dibit_vector.jl

```diff
@@ -0,0 +1,70 @@
+"""
+    DiBitVector <: AbstractVector{UInt8}
+
+A bitvector whose elements are two bits wide, allowing
+storage of integer values between 0 and 3. Optimized
+for performance and memory savings.
+"""
+mutable struct DiBitVector <: AbstractVector{UInt8}
+    data::Vector{UInt64}
+    len::UInt
+
+    function DiBitVector(n::Integer, v::Integer)
```

Comment:
I think a docstring for the constructors is warranted, in particular this one which is unlike what is available for other `AbstractVector`s. By the way, it might even appear backward with some of `Array` constructors, like in `Array{Union{Int,Nothing}}(nothing, 2)` where the initializer is given before the dimension.

Reply:
How do you write a docstring for an inner constructor? Do you just put it in the main struct docstring? (I tried doing it above the inner and it doesn't show up.)

Reply:
Got it.

```
help?> DiBitVector
search: DiBitVector
  DiBitVector(n::Integer, v::Integer)
  Create a DiBitVector with n elements preinitialized to a value v from 0 to 3, inclusive.
  A DiBitVector is a vector whose elements are two bits wide, allowing storage of integer values between 0 and 3. This structure is optimized for
  performance and memory savings for large numbers of elements.
  ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
  DiBitVector(n::Integer)
  Create a DiBitVector with n elements set to 0.
```

Reply:
Thinking about this a bit more... How likely it is that this will be extended to `DiBiArray` with multiple dimensions? In this case, putting the dimensions last would be more necessary (e.g. `DiBiArray(value, d1, d2)`, like for some arrays constructors, and `fill`, and `rand` etc.) So in any I would suggest to switch the order here in `DiBiVector` for consistency (but it's just my outsider's opinion).

Reply:
I have no plans on extending this to a matrix or n-dimentions. That said, I'm not averse to changing the constructor syntax. It means that we will have an inconsistency in position between
```
d = DiBitVector(2, 10)   # fill a 10-element DBV with values of 2
```
and
```
d = DiBitVector(10)  # zero-fill a 10-element DBV
```
but if that's the consensus I'll change it.

Reply:
Yes that's not ideal either. I'm not aware of whether there is a common practice in the package ecosystem for this case. If `DiBitVector(10)` is kept, I guess I prefer your current version for specifying the value, i.e. `DiBiVector(10, 2)`.

Reply:
`DiBitVector(10)` is useful; and I might even propose at some point that it allocate `undef` memory (since we always have `DiBitVector(10, 0)` to get zeroed memory).

Anyway, I'll mark this conversation resolved. We can always change things later based on wider feedback.